### ISSUE TITLE
restore from cloud backups

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/ts-android#502]
Especially since we are touting the Upload feature as a backup, we need to have a mechanism whereby someone can restore from their backup.

**Comment by neutrinog** [comment]
@jag3773 until we get the door43 accounts integrated into this process we can only provide backups on a device by device basis (not by user). Therefore, if they ever uninstall the app, get a different device, or manually regenerate their ssh keys (only available in the developer tools) they won't be able to access any past online backups.

**Comment by neutrinog** [comment]
@jag3773 when someone restores from the server how do you want that to operate? 
Here are some options:

1. the local repository is completely replaced by the one in the cloud. This will erase any outstanding local history.
1. just the `HEAD` of the backup in the cloud is copied into the local repository. this will retain any outstanding local history.

The benefit of the first method is they will be able to regain the entire history of the backup. Say for example they began some work, backed it up, deleted it, started again, but decided their backup was better.

The benefit of the second method is they will be able to retain the local history. Which is I think more likely going to be the case. Say for example I translate some stuff, backup it up, translate some more, then decide I want to restore from a backup, but then want to revert back to what I had previously translated between backing up and restoring. This method would allow them to do just that.

To sum up:
---
 it depends what history is more important to you. The backup in the cloud or the local history. I'm thinking it will be the local history.

I'm pretty sure there are some more complex solutions available to us in git that will allow us to maintain both, but that will be more time consuming to set up and ugly to maintain history-wise and I don't want to go that route unless we really need to.

**Comment by jag3773** [comment]
I think this is a great opportunity to use branches, right?

**Comment by neutrinog** [comment]
no because we could potentially be talking about two completely different repositories. Branching is fine if you are working with repositories that share the same history but if you have two different histories it becomes a lot more complex.

**Comment by jag3773** [comment]
Let's do this for Android:

* If they are restoring from the server and they have the same project existing on their device then grab `HEAD` from master and keep local history.
* If they are restoring from the server and they don't have the project locally then grab the whole thing.
